oxide-auth/src/primitives/registrar.rs

```diff
@@ -327,6 +390,77 @@ impl core::str::FromStr for ExactUrl {
     }
 }
 
+impl IgnoreLocalPortUrl {
+    /// Try to create a url that ignores port where host is localhost
+    pub fn new<'a, S: Into<Cow<'a, str>>>(url: S) -> Result<Self, ParseUrlError> {
+        let url: Cow<'a, str> = url.into();
+        let mut parsed: Url = url.parse()?;
+        match parsed.host_str() {
+            Some("localhost") => {
+                let _ = parsed.set_port(None);
+                Ok(IgnoreLocalPortUrl(IgnoreLocalPortUrlInternal::Local(parsed)))
+            }
+            _ => Ok(IgnoreLocalPortUrl(IgnoreLocalPortUrlInternal::Exact(
+                url.into_owned(),
+            ))),
+        }
```

Comment:
Can you use `From<Url>` here for consistency?

Reply:
If I replace that `match` with `Ok(IgnoreLocalPortUrl::from(parsed))`, so `IgnoreLocalPortUrl::new("http://example.com") == IgnoreLocalPortUrl(IgnoreLocalPortUrlInternal::Exact("http://127.0.0.1:8080/".to_string()))` (the normalized form of `"http://example.com"`).

That change can lead to strange behavior, like `RegisteredUrl::IgnorePortOnLocalhost(IgnoreLocalPortUrl::new("http://example.com"))` being different of `RegisteredUrl::Exact("http://example.com".to_string())`, but equal to `RegisteredUrl::Exact("http://example.com/".to_string())`.

Reply:
I see, there is a subtle difference with where `url.into_owned()` would become `parsed.into()` if changed. It irks me a bit that the main logic—the part that is also critical for fitness for purpose—of matching the host and unsetting the port is not obviously shared between those two. What a shame that `Url::parse().to_string()` does not roundtrip.

Okay, consider this solved then but feel free to add more comments explaining that subtlety in case a later programmer intends on changing and consolidating the two portions.